Automattic/jetpack
# Update check for classic vs. block theme in the WooCommerce Analytics module

## Proposed changes:

* Adds a check to determine if a classic theme is in use. That way we will only query the Cart/Checkout _pages_ and not the _templates_ when determining block vs. shortcode use and additional blocks used in the WooCommerce Analytics module.
* Updates the type of the `cart_page_contains_cart_block`, `cart_page_contains_cart_shortcode`, `checkout_page_contains_checkout_block`, and `checkout_page_contains_checkout_shortcode` properties so it is always a `0` or `1` rather than `true` or `false`.
* Checks for the presence of the `woocommerce/classic-shortcode` block as well as the normal shortcode
* Updates the check in `class-jetpack-woocommerce-analytics-universal.php` when getting additional blocks on a template to check for the template by its other possible name, then if still not found it bails cleanly without generating a warning.

### Other information:


## Jetpack product discussion



N/A

## Does this pull request change what data or activity we track or use?


No, it just updates the accuracy of the properties.

## Testing instructions:






*Please note, the properties for the Cart/Checkout Shortcode/Blocks use is fetched only once per day, and is stored in a transient, you will need to clear the `jetpack_woocommerce_analytics_cart_checkout_content_sources`, `jetpack_woocommerce_analytics_additional_blocks_on_checkout_page`, and `jetpack_woocommerce_analytics_additional_blocks_on_cart_page` transients while testing to ensure the most up to date data*

**Please set up a fresh site before testing**

Throughout testing, please check for PHP warnings. There should not be any.

### Classic theme testing (e.g. Storefront)

- Using a classic theme (e.g. Storefront) edit your Cart and Checkout pages, set them to be using the Cart and Checkout _blocks_.
- Add items to your cart and go to the Cart page.
- Using tracks vigilante (p7H4VZ-3cB-p2), ensure the following properties on the `woocommerceanalytics_cart_view` event contain:
  - `cart_page_contains_cart_block`: `1`
  - `cart_page_contains_cart_shortcode`: `0`
  - `checkout_page_contains_checkout_block`: `1`
  - `checkout_page_contains_checkout_shortcode`: `0`
- Proceed to the Checkout block and check the same properties and their values on the `woocommerceanalytics_product_checkout` and `woocommerceanalytics_checkout_view` events.
- Change your Cart/Checkout pages to use the shortcode block (enter `[woocommerce_cart]` on the Cart page, and `[woocommerce_checkout]` on the Checkout page).
- Proceed to the Checkout page again (note the `woocommerceanalytics_checkout_view` will not happen in a shortcode checkout, so don't expect it):
  - `cart_page_contains_cart_block`: `0`
  - `cart_page_contains_cart_shortcode`: `1`
  - `checkout_page_contains_checkout_block`: `0`
  - `checkout_page_contains_checkout_shortcode`: `1`

### Block theme testing (e.g. Twenty Twenty-Three)

- Switch to a block theme and repeat the above steps, however edit the Cart/Checkout templates rather than the pages.
- Install an incompatible extension (e.g. Checkout Field Editor).
- Delete *all* blocks from the pages before saving each time!
- Go to the Checkout template and select the Checkout block. In the block sidebar it will ask you if you'd like to revert to the classic Checkout.
<img width="273" alt="image" src="https://github.com/Automattic/jetpack/assets/5656702/87d723a1-15c1-42f0-afa9-3576f46eb5ab">

- Click that, and wait for the classic shortcode block to show, it looks like this:
<img width="670" alt="image" src="https://github.com/Automattic/jetpack/assets/5656702/8ebb3bcb-eb05-44ce-a1bb-b9bb8e44a689">

- Save your changes and do the same on the Cart template.
- Go to the Checkout block and ensure the properties on the `woocommerceanalytics_product_checkout` are:
  - `cart_page_contains_cart_block`: `0`
  - `cart_page_contains_cart_shortcode`: `1`
  - `checkout_page_contains_checkout_block`: `0`
  - `checkout_page_contains_checkout_shortcode`: `1`


### Page content on block themes

- Go to Appearance -> Editor -> Templates -> Page: Cart
- Click the three dots at the top and choose "Clear customizations"

<img width="467" alt="image" src="https://github.com/Automattic/jetpack/assets/5656702/85ca2cd6-26eb-42e1-95bc-e0ffddb61e37">

- If the dots aren't there, make some changes to the template, save it, then close the tab and reopen the editor in a new tab.

- Go to Appearance -> Editor -> Templates -> Page: Checkout
- Click the three dots at the top and choose "Clear customizations" like above.

Check that both templates are using the "Page content" block. (If you edit the template and open the list view, it should look like this:)
<img width="452" alt="image" src="https://github.com/Automattic/jetpack/assets/5656702/ae3ff34a-3e4a-4980-9b44-5cb84d263df8">

- Go back to the Admin dashboard front page. Repeat the testing steps from **Classic theme testing (e.g. Storefront)**
